### ISSUE TITLE
Add extensions support

### ISSUE THREAD

**Issue by aloisdg** [mailpile/Mailpile#1579]
Hello,

I would love if we can add an extension to our mail.

If the subject of the mail has an extension, then:

- "Foobar is here"->classic text/html render
- "Foobar is here.md"->markdown render
- "Foobar is here.adoc"->asciidoc render [caveat](https://www.drupal.org/node/2595473)
- "Foobar is here.tex"->latex render

I found that the support of Markdown has already been ask at #569.

What do you think of this?

**Comment by BjarniRunar** [comment]
E-mail already has a standard for doing this: MIME types. Extensions are in every way inferior. :-)

That said, I think the main point of your request is to allow the author to write rich content in their preferred tool and have the receiving end interpret that. I understand the appeal!

Unfortunately, adding support for all of those renderings is a lot of work, with some very serious security implications. Every single added format is a potential attack vector (see https://stackoverflow.com/questions/4052686/latex-security-considerations for an example), and doing this dramatically increase the risk of automated remote exploitation of Mailpile.

For those reasons, I don't see this happening in the near future and I would even reject pull requests for features like this - once our plugin APIs mature a bit folks might be able to write plugins that do this, but I don't see this kind of thing in Mailpile core.

Adding support for composition-time conversion, that converts Markdown or Asciidoc or even LaTeX to HTML and plain-text would not have the same security implications. I would encourage people to consider this instead of changing the format of e-mail and complicating the receiving end.

Thanks for filing! :-)

**Comment by cbz20** [comment]
> Adding support for composition-time conversion, that converts Markdown or Asciidoc or even LaTeX to HTML and plain-text would not have the same security implications. I would encourage people to consider this instead of changing the format of e-mail and complicating the receiving end.

For rendering LaTeX code in html, there is a nice Javascript library called [MathJax](https://www.mathjax.org/). By using a local installation of MathJax and choosing the right configuration, this should be pretty safe.

Actually, I have already written some MathJax code for Mailpile. ​So far, the features are quite basic: Firstly, an on/off button for displaying maths in email conversations, and secondly, a field below the email composer which displays a preview of the email with typeset maths. This preview gets updated as the email is being written (onkeyup):

![mathjaxformailpilecomments](https://cloud.githubusercontent.com/assets/9016969/13725487/4507bcc2-e89b-11e5-82e2-d3b0886fc578.PNG)

However, I did this before the major changes to the compose/display layout last autumn. I decided to wait until the first stable version is released, before I look into it again, but I'd be happy to!
